Build failure: dotnetCorePackages.dotnet_10.vmr

Forked from https://github.com/NixOS/nixpkgs/issues/430989#issuecomment-3157180482

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

build dotnetCorePackages.dotnet_10.vmr on x86_64-linux

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
error: builder for '/nix/store/37v3zh25741cqykcg4npnsb4y31i5phg-dotnet-vmr-10.0.0-preview.6.drv' failed with exit code 139;
       last 25 log lines:
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Updating Autotools / GNU config script to a newer upstream version: ./src/runtime/src/native/external/zlib-ng/tools/config.sub
       > Running phase: configurePhase
       >
       > Welcome to .NET 10.0!
       > ---------------------
       > SDK Version: 10.0.100-preview.6.25358.103
       >
       > ----------------
       > Installed an ASP.NET Core HTTPS development certificate.
       > To trust the certificate, run 'dotnet dev-certs https --trust'
       > Learn about HTTPS: https://aka.ms/dotnet-https
       >
       > ----------------
       > Write your first app: https://aka.ms/dotnet-hello-world
       > Find out what's new: https://aka.ms/dotnet-whats-new
       > Explore documentation: https://aka.ms/dotnet-docs
       > Report issues and find source on GitHub: https://github.com/dotnet/core
       > Use 'dotnet --help' to see available commands or visit: https://aka.ms/dotnet-cli
       > --------------------------------------------------------------------------------------
       > Package source with Name: Package source 1 added successfully.
       > Package source with Name: Package source 2 added successfully.
       >   ./.dotnet SDK directory exists...it will not be installed
       >
       > ./prep-source-build.sh: line 291:  6140 Segmentation fault         (core dumped) "$dotnetSdk/dotnet" build "$REPO_ROOT/eng/init-detect-binaries.proj" "/p:BinariesMode=Clean" "/p:AllowedBinariesFile=$REPO_ROOT/eng/allowed-sb-binaries.txt" "/p:BinariesPackagesDir=$packagesDir" "/bl:artifacts/log/prep-remove-binaries.binlog" "/fileLoggerParameters:LogFile=artifacts/log/prep-remove-binaries.log" "${positional_args[@]}"
       For full logs, run 'nix log /nix/store/37v3zh25741cqykcg4npnsb4y31i5phg-dotnet-vmr-10.0.0-preview.6.drv'.
error: 1 dependencies of derivation '/nix/store/kcgz90gj2apq2axk4l38bligg8k1g1p6-dotnet-sdk-10.0.100-preview.6.25358.103.drv' failed to build
error: 1 dependencies of derivation '/nix/store/an8kxgyi8wjnd2mg7fhjzdcfsi27r3mq-dotnet-sdk-wrapped-10.0.100-preview.6.25358.103.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3hr2sj9qcvg8v4fkyik9jx5gsh5mcz60-nix-shell-env.drv' failed to build
```

### Additional context

_No response_

### System metadata

@michaelhthomas can you provide this? (`nix-shell -p nix-info --run "nix-info -m"`)

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

I am getting build errors that may be related to this issue

```txt

...  more Installing here ...

-- Installing: /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/./libsuperpmi-shim-collector.so
    -- Installing: /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/./libsuperpmi-shim-counter.so
    -- Installing: /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/./libsuperpmi-shim-simple.so
    /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/src/coreclr
    Repo successfully built.
    Product binaries are available at /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release
    ILCompiler.DependencyAnalysisFramework -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler.DependencyAnalysisFramework/x64/Release/ILCompiler.DependencyAnalysisFramework.dll
    ILCompiler.TypeSystem -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler.TypeSystem/x64/Release/ILCompiler.TypeSystem.dll
    ILCompiler.MetadataTransform -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler.MetadataTransform/x64/Release/ILCompiler.MetadataTransform.dll
    ILCompiler.Compiler -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler.Compiler/x64/Release/ILCompiler.Compiler.dll
    ILCompiler.RyuJit -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler.RyuJit/x64/Release/ILCompiler.RyuJit.dll
    ILCompiler_inbuild -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler_inbuild/x64/Release/ilc.dll
    Optimizing assemblies for size. This process might take a while.
    ILCompiler_inbuild -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/x64/ilc/
    ILCompiler.Diagnostics -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler.Diagnostics/x64/Release/ILCompiler.Diagnostics.dll
    ILCompiler.ReadyToRun -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler.ReadyToRun/x64/Release/ILCompiler.ReadyToRun.dll
    crossgen2_inbuild -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/crossgen2_inbuild/x64/Release/crossgen2.dll
    Optimizing assemblies for size. This process might take a while.
    crossgen2_inbuild -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/x64/crossgen2/
    DownlevelLibraryImportGenerator -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/DownlevelLibraryImportGenerator/Release/netstandard2.0/Microsoft.Interop.LibraryImportGenerator.Downlevel.dll
    ILCompiler.Build.Tasks -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/ilc-published/netstandard/ILCompiler.Build.Tasks.dll
    ILCompiler_publish -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/ILCompiler_publish/x64/Release/ilc.dll
    Generating native code
    ILCompiler_publish -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/ilc-published/
    crossgen2_publish -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/crossgen2_publish/x64/Release/crossgen2.dll
    Generating native code
    crossgen2_publish -> /build/dotnet-10.0.100-preview.7.25380.108/src/runtime/artifacts/bin/coreclr/linux.x64.Release/crossgen2-published/
  /build/dotnet-10.0.100-preview.7.25380.108/.dotnet/sdk/10.0.100-preview.7.25380.108/Microsoft.Common.CurrentVersion.targets(5381,5): error MSB3026: Could not copy "
  'runtime' failed during build.
  See '/build/dotnet-10.0.100-preview.7.25380.108/artifacts/log/Release/runtime/runtime.log' for more information.

Build FAILED.

/build/dotnet-10.0.100-preview.7.25380.108/repo-projects/Directory.Build.targets(598,5): error MSB3073: The command "/build/dotnet-10.0.100-preview.7.25380.108/src/runtime/build.sh --restore --build --pack --publish  --sourceBuild --fromVMR --configuration Release --verbosity minimal /p:TargetRid=linux-x64 /p:OfficialBuildId=20250730.108 /p:DotNetPackageVersionPropsPath=/build/dotnet-10.0.100-preview.7.25380.108/artifacts/obj/PackageVersions/PackageVersions.runtime.props /p:SourceBuiltAssetsDir=/build/dotnet-10.0.100-preview.7.25380.108/artifacts/assets/Release/ /p:SourceBuiltAssetManifestsDir=/build/dotnet-10.0.100-preview.7.25380.108/artifacts/obj/manifests/Release/runtime/ /p:RepositoryName=runtime /p:RepositoryUrl=https://github.com/dotnet/dotnet /p:PreviouslySourceBuiltNupkgCacheDir="/nix/store/4hs7kdxkshfgi6hqzm33xwr0w8g893g2-dotnet-stage0-sdk-10.0.100-preview.7.25380.108-artifacts/" /p:ReferencePackageNupkgCacheDir="/build/dotnet-10.0.100-preview.7.25380.108/prereqs/packages/reference/" /p:TrackPrebuiltUsageReportDir="/build/dotnet-10.0.100-preview.7.25380.108/artifacts/log/Release/runtime/" /p:SourceBuiltShippingPackagesDir=/build/dotnet-10.0.100-preview.7.25380.108/artifacts/packages/Release/Shipping/runtime/ /p:SourceBuiltNonShippingPackagesDir=/build/dotnet-10.0.100-preview.7.25380.108/artifacts/packages/Release/NonShipping/runtime/ /p:SourceBuiltPdbArtifactsDir=/build/dotnet-10.0.100-preview.7.25380.108/artifacts/SymStore/Release/runtime/ /p:SourceBuiltArtifactsBasePath=/build/dotnet-10.0.100-preview.7.25380.108/ /p:UsingToolMicrosoftNetCompilers=false /p:PortableBuild=true /p:PortableTargetRid=linux-x64 /p:RepoOriginalSourceRevisionId=46be99d54b6d3fcd027d87f697cd4a7a8658111a  -bl --arch x64 --os linux --bootstrap /p:DotNetBuildAllRuntimePacks=true > /build/dotnet-10.0.100-preview.7.25380.108/artifacts/log/Release/runtime/runtime.log 2>&1" exited with code 1. [/build/dotnet-10.0.100-preview.7.25380.108/repo-projects/runtime.proj]
    0 Warning(s)
    1 Error(s)
```

--- a/pkgs/development/compilers/dotnet/vmr.nix
+++ b/pkgs/development/compilers/dotnet/vmr.nix
@@ -2,6 +2,7 @@
   clangStdenv,
   lib,
   fetchurl,
+  fetchpatch,
   dotnetCorePackages,
   jq,
   curl,
@@ -146,6 +147,12 @@ stdenv.mkDerivation rec {
     ]
     ++ lib.optionals (lib.versionAtLeast version "10") [
       ./bundler-fix-file-size-estimation-when-bundling-symli.patch
+      (fetchpatch {
+        url = "https://github.com/dotnet/runtime/commit/118eacc4f40f1ef48b47c0b7ff40ac0b3ae8c28a.patch";
+        hash = "sha256-5sRGEpULAgjDjU1LKm7Pcx3Qfbr891CB9apOpYdzPyA=";
+        stripLen = 1;
+        extraPrefix = "src/runtime/";
+      })
     ];
 
   postPatch = ''

